File: task-03/Elixir/Sub task 3/subtask3.exs
defmodule Diamond do
  def print_diamond(n) when n < 1 do
    IO.puts("Number should be greater than 1")
  end

  def print_diamond(n) do
    mid = div(n, 2)

    # Upper part of the diamond
    for i <- 1..n//2 + 1, rem(2 * i - 1, 2) == 1 do
      spaces = mid - i + 1
      stars = 2 * i - 1
      IO.puts String.duplicate(" ", spaces) <> String.duplicate("*", stars)
    end

    # Lower part of the diamond
    if rem(n, 2) == 0 do
      for i <- n//2..1, rem(2 * i - 1, 2) == 1 do
        spaces = mid - i + 1
        stars = 2 * i - 1
        IO.puts String.duplicate(" ", spaces) <> String.duplicate("*", stars)
      end
    else
      for i <- n//2..1, rem(2 * i - 1, 2) == 1 do
        spaces = mid - i + 1
        stars = 2 * i - 1
        IO.puts String.duplicate(" ", spaces) <> String.duplicate("*", stars)
      end
    end
  end
end

IO.write("Enter the number of rows for the diamond: ")
n = IO.gets("") |> String.trim() |> String.to_integer()
Diamond.print_diamond(n)

File: task-03/Ruby/Sub task 3/subtask3.rb
def print_diamond(n)
  if n < 1
    puts "Number should be greater than 1"
    return
  end

  mid = n / 2

  # Upper part of the diamond
  (1..n).step(2) do |i|
    # Calculate and print leading spaces
    spaces = mid - i / 2
    puts ' ' * spaces + '*' * i
  end

  # Lower part of the diamond
  if n.even?
    (n - 1).step(1, -2) do |i|
      # Calculate and print leading spaces with a slight shift for asymmetry
      spaces = mid - i / 2 + 1
      puts ' ' * (spaces - 1) + '*' * i
    end
  else
    (n - 2).step(1, -2) do |i|
      spaces = mid - i / 2
      puts ' ' * spaces + '*' * i
    end
  end
end

# Example usage
print "Enter the number of rows for the diamond: "
n = gets.to_i
print_diamond(n)

